### PR TITLE
Add honorific_prefix field

### LINKED ISSUE #2: Handle Dr. honorific prefixes
Multiple members in the live data have Dr. attached to their name. The scraper needs to strip this title from the name and return it as an honorific prefix.

### PR DESCRIPTION
**Pull request by ondenman**
This is a step towards fixing: https://github.com/everypolitician/everypolitician-data/issues/25705

A member is incoming with Prof. attached to their name. This PR adds an honorific prefix field.

This PR only knows of one title: 'Prof'. There a multiple members in the live data with 'Dr.' attached to their names. A future PR will add Dr. to the list of prefixes. (Fixes: https://github.com/everypolitician-scrapers/tanzania-parliament/issues/2)

~**Note**: There is no test associated with this change as I cannot find a straightforward way to test it. The data is captured from a MemberRow, and the members list is paginated.~

### CODE REVIEW

**Review by tmtmtmtm — CHANGES_REQUESTED** [review]
I think you definitely need to add a test for this.

I'm not quite sure why the list page being paginated would make a difference here — surely `MembersPage` can be instantiated with the URL to any of the relevant pages?

File: lib/member_row.rb [hunk 1]
```diff
@@ -32,6 +36,14 @@ class MemberRow < Scraped::HTML
 
   private
 
+  def prefixes
+    %w(Prof.)
+  end
+
+  def name_parts
+    noko[1].text.strip.sub(/^Hon.? /, '').split(' ')
```
Comment (tmtmtmtm) [inline lib/member_row.rb]: Let's be consistent how we did this in Malta, and have an "unwanted" list with 'Hon'. The more we can do this the same way across multiple scrapers, the easier it will be to discover what the _real_ edge-cases are. Doing things differently in each scraper makes it much harder to extract the commonalities out to a library.

File: test/member_data_test.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+require_relative '../lib/members_page.rb'
+
+describe 'member data' do
+  url = 'http://www.parliament.go.tz/mps-list'
+  around { |test| VCR.use_cassette(url.split('/').last, &test) }
+
+  subject do
+    MembersPage.new(response: Scraped::Request.new(url: url).response)
+               .member_rows
+               .select { |row| row.to_h[:id] == id }.first
```
Comment (tmtmtmtm) [inline test/member_data_test.rb]: As with Sint Maarten: `.select { … }.first` should be `.find`, and `row.to_h[:id]` could just be `row.id`

**Review by tmtmtmtm — CHANGES_REQUESTED** [review]

File: lib/member_row.rb [hunk 3]
```diff
@@ -32,6 +36,22 @@ class MemberRow < Scraped::HTML
...
+  def unwanted_prefixes
+    %w(Hon.)
+  end
+
+  def prefixes
+    wanted_prefixes + unwanted_prefixes
+  end
+
+  def name_parts
+    noko[1].text.strip.split(' ')
```
Comment (tmtmtmtm) [inline lib/member_row.rb]: we generally use `.tidy` rather than `.strip`, as it takes care of more edge cases

File: lib/member_row.rb [hunk 4]
```diff
@@ -7,11 +7,15 @@ class MemberRow < Scraped::HTML
   end
 
   field :photo do
-    URI.encode(noko[0].css('img/@src').text)
+    noko[0].css('img/@src').text.gsub(' ', '%20')
```
Comment (tmtmtmtm) [inline lib/member_row.rb]: This should be in a separate PR, as it's not directly connected to adding prefixes.

**Review by tmtmtmtm — CHANGES_REQUESTED** [review]

**Comment by ondenman** [comment]
The associated issue in ep-data still has steps remaining.

**Comment by tmtmtmtm** [comment]
I've resolved the conflict, rebased this over master, and fixed the broken test.

**Review by tmtmtmtm — APPROVED** [review]
